workspace/test_configurations.go

```diff
@@ -0,0 +1,89 @@
+package workspace
+
+import (
+	"runtime"
+	"strings"
+)
+
+type TestConfiguration struct {
+	// The static portion of the test Command, which will be run for every test on this track. Examples include `cargo test` or `go test`.
+	// Might be empty if there are platform-specific versions
+	Command string
+
+	// Windows-specific test command. Mostly relevant for tests wrapped by shell invocations. Falls back to `Command` if we're not running windows or this is empty.
+	WindowsCommand string
+}
+
+func (c *TestConfiguration) GetTestCommand() (string, error) {
+	var cmd string
+	if runtime.GOOS == "windows" && c.WindowsCommand != "" {
+		cmd = c.WindowsCommand
+	} else {
+		cmd = c.Command
+	}
+
+	if strings.Contains(cmd, "{{test_files}}") {
+		testFiles, err := getTestFiles()
+		if err != nil {
+			return "", err
+		}
+		cmd = strings.ReplaceAll(cmd, "{{test_files}}", strings.Join(testFiles, " "))
+	}
+
+	return cmd, nil
+}
+
+var TestConfigurations = map[string]TestConfiguration{
+	"8th": {
+		Command:        "tester.sh",
+		WindowsCommand: "tester.bat",
+	},
+	"ballerina": {
+		Command: "bal test",
+	},
+	"c": {
+		Command: "make",
+	},
+	"cfml": {
+		Command: "box task run TestRunner",
+	},
+	"cobol": {
+		Command:        "test.sh",
+		WindowsCommand: "test.ps1",
+	},
+	"coffeescript": {
+		Command: "jasmine-node --coffee {{test_files}}",
```

Comment:
I experimented with putting `{{test_files}}` in a const (which would improve the replacement code), but it made writing the commands less clear (one of `"ruby " + TEST_FiLES` or `fmt.Sprintf("ruby %s", TEST_FILES)` or `injectTestFilesPlaceholder("ruby %s")`, all of which seemed to defeat the point). 

As written, there's a higher chance that someone adding a new command will typo the placeholder and the replacement won't happen, but it provides the most clarity when reading the actual commands.

Reply:
I think this is nice!